eventstore/src/main/java/com/github/yafna/events/pipelines/EventPipeline.java

```diff
@@ -0,0 +1,73 @@
+package com.github.yafna.events.pipelines;
+
+import com.github.yafna.events.DomainEvent;
+import com.github.yafna.events.EmittedEvent;
+import com.github.yafna.events.Event;
+import com.github.yafna.events.EventMeta;
+import com.github.yafna.events.aggregate.PayloadUtils;
+import com.github.yafna.events.dispatcher.EventDispatcher;
+import com.github.yafna.events.handlers.EventHandler;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Stream;
+
+@Slf4j
+public class EventPipeline<A, T extends DomainEvent<A>> {
+    private final ForkJoinPool executor;
+    private final EventDispatcher dispatcher;
+    private final String origin;
+    private final String handlerId;
+    private final EventHandler<T> handler;
+
+    public EventPipeline(
+            EventDispatcher dispatcher, Class<T> eventType, EventHandler<T> handler, String handlerId, Clock clock
+    ) {
+        this(dispatcher, eventType, handler, handlerId, clock, new ForkJoinPool());
+    }
+
+    public EventPipeline(
+            EventDispatcher dispatcher, Class<T> eventType, EventHandler<T> handler, String handlerId, Clock clock, ForkJoinPool executor
+    ) {
+        origin = PayloadUtils.origin(eventType);
+        this.executor = executor;
+        this.dispatcher = dispatcher;
+        this.handler = handler;
+        this.handlerId = handlerId;
+        Duration timeWindow = Duration.ofDays(1);
+        Instant since = clock.instant().minus(timeWindow);
```

Comment:
A default value for time window. Has to be made a constructor parameter along with adding some tests